File: 4212-ToggleLightBulbs/4212-ToggleLightBulbs.cpp
// Last updated: 3/30/2026, 2:10:40 PM
class Solution {
public:
    vector<int> toggleLightBulbs(vector<int>& bulbs) {
        set<int> uniquebulbs;
        unordered_map<int,int> hashmap;
        for(int bulb : bulbs) hashmap[bulb]++;
        for(int bulb : bulbs) if(hashmap[bulb]%2!=0) uniquebulbs.insert(bulb);
        vector<int> result(uniquebulbs.begin(),uniquebulbs.end());
        sort(result.begin(),result.end());
        return result;
    }
};

File: 4212-ToggleLightBulbs/4212-ToggleLightBulbs.java
// Last updated: 3/30/2026, 2:10:46 PM
class Solution {
    public List<Integer> toggleLightBulbs(List<Integer> bulbs) {
        HashMap<Integer,Integer> countarr = new HashMap<>();
        ArrayList<Integer> values = new ArrayList<Integer>();
        for(int bulb: bulbs)
        {
            countarr.put(bulb,countarr.getOrDefault(bulb,0)+1);    
        }
        for(int key: countarr.keySet())
        {
            if(countarr.get(key)%2!=0) values.add(key);
        }
        Collections.sort(values);
        return values;
    }
}